tiflash crash for the error of invalid flag l in lock value

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
```
{"namespace":"e2e-htap-fast-mode-tps-1303065-1-303","log":"[2022/11/03 01:53:16.688 +08:00] [ERROR] [BaseDaemon.cpp:570] [\"\\n       0x55269b1\\tfaultSignalHandler(int, siginfo_t*, void*) [tiflash+89287089]\\n                \\tlibs/libdaemon/src/BaseDaemon.cpp:221\\n  0x7f7c84dee630\\t<unknown symbol> [libpthread.so.0+63024]\\n       0x60b6482\\tDB::isUnsupportedEncodeType(std::__1::vector<tipb::FieldType, std::__1::allocator<tipb::FieldType> > const&, tipb::EncodeType) [tiflash+101409922]\\n                \\tdbms/src/Flash/Coprocessor/DAGUtils.cpp:1172\\n       0x60977b5\\tDB::DAGContext::initOutputInfo() [tiflash+101283765]\\n                \\tdbms/src/Flash/Coprocessor/DAGContext.cpp:52\\n       0x61818e7\\tDB::MPPTask::prepare(mpp::DispatchTaskRequest const&) [tiflash+102242535]\\n                \\tdbms/src/Flash/Mpp/MPPTask.cpp:271\\n       0x617865d\\tDB::MPPHandler::execute(std::__1::shared_ptr<DB::Context> const&, mpp::DispatchTaskResponse*) [tiflash+102205021]\\n                \\tdbms/src/Flash/Mpp/MPPHandler.cpp:57\\n       0x6072e8a\\tDB::FlashService::DispatchMPPTask(grpc::ServerContext*, mpp::DispatchTaskRequest const*, mpp::DispatchTaskResponse*) [tiflash+101133962]\\n                \\tdbms/src/Flash/FlashService.cpp:210\\n       0x67ae629\\tgrpc::internal::RpcMethodHandler<tikvpb::Tikv::Service, mpp::DispatchTaskRequest, mpp::DispatchTaskResponse, google::protobuf::MessageLite, google::protobuf::MessageLite>::RunHandler(grpc::internal::MethodHandler::HandlerParameter const&) [tiflash+108717609]\\n                \\tcontrib/grpc/include/grpcpp/impl/codegen/method_handler.h:113\\n       0x67e4700\\tgrpc::Server::SyncRequest::ContinueRunAfterInterception() [tiflash+108939008]\\n                \\tcontrib/grpc/src/cpp/server/[server_cc.cc:433](http://server_cc.cc:433/)\\n       0x67e4531\\tgrpc::Server::SyncRequest::Run(std::__1::shared_ptr<grpc::Server::GlobalCallbacks> const&, bool) [tiflash+108938545]\\n                \\tcontrib/grpc/src/cpp/server/[server_cc.cc:421](http://server_cc.cc:421/)\\n       0x67f6725\\tgrpc::ThreadManager::WorkerThread::WorkerThread(grpc::ThreadManager*)::$_0::__invoke(void*) [tiflash+109012773]\\n                \\tcontrib/grpc/src/cpp/thread_manager/[thread_manager.cc:36](http://thread_manager.cc:36/)\\n       0x6c33c0a\\tgrpc_core::(anonymous namespace)::ThreadInternalsPosix::ThreadInternalsPosix(char const*, void (*)(void*), void*, bool*, grpc_core::Thread::Options const&)::'lambda'(void*)::__invoke(void*) [tiflash+113458186]\\n                \\tcontrib/grpc/src/core/lib/gprpp/[thd_posix.cc:110](http://thd_posix.cc:110/)\\n  0x7f7c84de6ea5\\tstart_thread [libpthread.so.0+32421]\"] [source=BaseDaemon] [thread_id=870]\n","stream":"stdout","container":"serverlog","pod":"tc-tiflash-1","time":"2022-11-02T17:53:16.823897476Z"}
2022-11-03 01:53:16	
{"namespace":"e2e-htap-fast-mode-tps-1303065-1-303","log":"[2022/11/03 01:53:16.681 +08:00] [DEBUG] [DiagnosticsService.cpp:170] [\"start to search file /data0/logs/server.log\"] [source=DiagnosticsService] [thread_id=81]\n","stream":"stdout","container":"serverlog","pod":"tc-tiflash-1","time":"2022-11-02T17:53:16.823893024Z"}
2022-11-03 01:53:16	
{"namespace":"e2e-htap-fast-mode-tps-1303065-1-303","log":"[2022/11/03 01:53:16.666 +08:00] [DEBUG] [KVStore.cpp:383] [\"[region 772] approx mem cache info: rows 761, bytes 99163\"] [thread_id=62]\n","stream":"stdout","container":"serverlog","pod":"tc-tiflash-1","time":"2022-11-02T17:53:16.8238892Z"}
2022-11-03 01:53:16	
{"namespace":"e2e-htap-fast-mode-tps-1303065-1-303","log":"[2022/11/03 01:53:16.638 +08:00] [ERROR] [BaseDaemon.cpp:423] [\"Address not mapped to object.\"] [source=BaseDaemon] [thread_id=870]\n","stream":"stdout","container":"serverlog","pod":"tc-tiflash-1","time":"2022-11-02T17:53:16.823885359Z"}
2022-11-03 01:53:16	
{"namespace":"e2e-htap-fast-mode-tps-1303065-1-303","log":"[2022/11/03 01:53:16.638 +08:00] [ERROR] [BaseDaemon.cpp:414] [\"Access: read.\"] [source=BaseDaemon] [thread_id=870]\n","stream":"stdout","container":"serverlog","pod":"tc-tiflash-1","time":"2022-11-02T17:53:16.823881379Z"}
2022-11-03 01:53:16	
{"namespace":"e2e-htap-fast-mode-tps-1303065-1-303","log":"[2022/11/03 01:53:16.638 +08:00] [ERROR] [BaseDaemon.cpp:408] [\"Address: 0x8\"] [source=BaseDaemon] [thread_id=870]\n","stream":"stdout","container":"serverlog","pod":"tc-tiflash-1","time":"2022-11-02T17:53:16.823873414Z"}
2022-11-03 01:53:16	
{"namespace":"e2e-htap-fast-mode-tps-1303065-1-303","log":"[2022/11/03 01:53:16.638 +08:00] [ERROR] [BaseDaemon.cpp:378] [\"(from thread 505) Received signal Segmentation fault(11).\"] [source=BaseDaemon] [thread_id=870]\n","stream":"stdout","container":"serverlog","pod":"tc-tiflash-1","time":"2022-11-02T17:53:16.823869585Z"}
2022-11-03 01:53:16	
{"namespace":"e2e-htap-fast-mode-tps-1303065-1-303","log":"[2022/11/03 01:53:16.638 +08:00] [ERROR] [BaseDaemon.cpp:377] [########################################] [source=BaseDaemon] [thread_id=870]\n","stream":"stdout","container":"serverlog","pod":"tc-tiflash-1","time":"2022-11-02T17:53:16.823865814Z"}


{"namespace":"e2e-htap-fast-mode-tps-1303065-1-303","log":"[2022/11/03 01:54:38.330 +08:00] [ERROR] [Exception.cpp:89] [\"Code: 49, e.displayText() = DB::Exception: invalid flag l in lock value 534A748000000000000A1F5F6980000000000000010380000000000000430380000000000000058780C0C0F3BCB98806CB9C01660610E5E738100007630610E5E7381000086C06000E96921C000901, e.what() = DB::Exception, Stack trace:\\n\\n\\n       0x1608cde\\tDB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) [tiflash+23104734]\\n                \\tdbms/src/Common/Exception.h:46\\n       0x5e9e3b6\\tDB::RegionCFDataBase<DB::RegionLockCFDataTrait>::insert(DB::StringObject<true>&&, DB::StringObject<false>&&) [tiflash+99214262]\\n                \\tdbms/src/Storages/Transaction/RegionCFDataBase.cpp:56\\n       0x5e940de\\tauto DB::Region::handleWriteRaftCmd(DB::WriteCmdsView const&, unsigned long, unsigned long, DB::TMTContext&)::$_12::operator()<unsigned long>(unsigned long) const [tiflash+99172574]\\n                \\tdbms/src/Storages/Transaction/Region.cpp:650\\n       0x5e93122\\tDB::Region::handleWriteRaftCmd(DB::WriteCmdsView const&, unsigned long, unsigned long, DB::TMTContext&) [tiflash+99168546]\\n                \\tdbms/src/Storages/Transaction/Region.cpp:716\\n       0x5e58a5e\\tDB::KVStore::handleWriteRaftCmd(DB::WriteCmdsView const&, unsigned long, unsigned long, unsigned long, DB::TMTContext&) [tiflash+98929246]\\n                \\tdbms/src/Storages/Transaction/KVStore.cpp:293\\n       0x5e77fa5\\tHandleWriteRaftCmd [tiflash+99057573]\\n                \\tdbms/src/Storages/Transaction/ProxyFFI.cpp:95\\n  0x7fee98b2e0e8\\t_$LT$engine_store_ffi..observer..TiFlashObserver$u20$as$u20$raftstore..coprocessor..QueryObserver$GT$::post_exec_query::hb5bf0b2c254325c7 [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+17461480]\\n  0x7fee99b1bdcd\\traftstore::store::fsm::apply::ApplyDelegate$LT$EK$GT$::apply_raft_cmd::hf7a6d6155ca2451a [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+34164173]\\n  0x7fee99b30f23\\traftstore::store::fsm::apply::ApplyDelegate$LT$EK$GT$::process_raft_cmd::ha30c296b18e1c6bd [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+34250531]\\n  0x7fee99b333a1\\traftstore::store::fsm::apply::ApplyDelegate$LT$EK$GT$::handle_raft_committed_entries::h5799bec4fdd531d6 [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+34259873]\\n  0x7fee99b0e3bd\\traftstore::store::fsm::apply::ApplyFsm$LT$EK$GT$::handle_apply::h974b8d4bd8e245fe [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+34108349]\\n  0x7fee99b11979\\traftstore::store::fsm::apply::ApplyFsm$LT$EK$GT$::handle_tasks::hfdba29a9719d1fa6 [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+34122105]\\n  0x7fee990e3aab\\t_$LT$raftstore..store..fsm..apply..ApplyPoller$LT$EK$GT$$u20$as$u20$batch_system..batch..PollHandler$LT$raftstore..store..fsm..apply..ApplyFsm$LT$EK$GT$$C$raftstore..store..fsm..apply..ControlFsm$GT$$GT$::handle_normal::h6cb15547838608f1 [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+23448235]\\n  0x7fee9907c417\\tbatch_system::batch::Poller$LT$N$C$C$C$Handler$GT$::poll::h204060a2c1c34dd7 [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+23024663]\\n  0x7fee99133622\\tstd::sys_common::backtrace::__rust_begin_short_backtrace::h6193f1063b323c69 [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+23774754]\\n  0x7fee991680c9\\tcore::ops::function::FnOnce::call_once$u7b$$u7b$vtable.shim$u7d$$u7d$::h7d3b82de30c485a3 [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+23990473]\\n  0x7fee9a1de485\\tstd::sys::unix::thread::Thread::new::thread_start::h501a801cbc05471d [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+41251973]\\n                \\t/rustc/0f4bcadb46006bc484dad85616b484f93879ca4e/library/std/src/sys/unix/[thread.rs:108](http://thread.rs:108/)\\n  0x7fee97872ea5\\tstart_thread [libpthread.so.0+32421]\\n  0x7fee96c7796d\\t__clone [libc.so.6+1042797]\"] [source=\"DB::EngineStoreApplyRes DB::HandleWriteRaftCmd(const DB::EngineStoreServerWrap *, DB::WriteCmdsView, DB::RaftCmdHeader)\"] [thread_id=52]\n","pod":"tc-tiflash-1","container":"errorlog","stream":"stdout","time":"2022-11-02T17:54:39.180006645Z"}
```
### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiFlash version? (Required)
master

--- a/dbms/src/Storages/Transaction/RegionCFDataBase.cpp
+++ b/dbms/src/Storages/Transaction/RegionCFDataBase.cpp
@@ -346,6 +346,12 @@ inline void decodeLockCfValue(DecodedLockCFValue & res)
                 }
                 break;
             }
+            case LAST_CHANGE_PREFIX:
+            {
+                UInt64 _last_change_ts = readUInt64(data, len);
+                UInt64 _versions_to_last_change = readVarUInt(data, len);
+                break;
+            }
             default:
             {
                 std::string msg = std::string("invalid flag ") + flag + " in lock value " + value.toDebugString();

--- a/dbms/src/Storages/Transaction/RegionCFDataBase.cpp
+++ b/dbms/src/Storages/Transaction/RegionCFDataBase.cpp
@@ -349,10 +349,10 @@ inline void decodeLockCfValue(DecodedLockCFValue & res)
             case LAST_CHANGE_PREFIX:
             {
                 // Used to accelerate TiKV MVCC scan, useless for TiFlash.
-                UInt64 _last_change_ts = readUInt64(data, len);
-                UInt64 _versions_to_last_change = readVarUInt(data, len);
-                UNUSED(_last_change_ts);
-                UNUSED(_versions_to_last_change);
+                UInt64 last_change_ts = readUInt64(data, len);
+                UInt64 versions_to_last_change = readVarUInt(data, len);
+                UNUSED(last_change_ts);
+                UNUSED(versions_to_last_change);
                 break;
             }
             default:

--- a/dbms/src/Storages/Transaction/RegionCFDataBase.cpp
+++ b/dbms/src/Storages/Transaction/RegionCFDataBase.cpp
@@ -348,8 +348,11 @@ inline void decodeLockCfValue(DecodedLockCFValue & res)
             }
             case LAST_CHANGE_PREFIX:
             {
+                // Used to accelerate TiKV MVCC scan, useless for TiFlash.
                 UInt64 _last_change_ts = readUInt64(data, len);
                 UInt64 _versions_to_last_change = readVarUInt(data, len);
+                UNUSED(_last_change_ts);
+                UNUSED(_versions_to_last_change);
                 break;
             }
             default:

--- a/dbms/src/Storages/Transaction/RegionCFDataBase.cpp
+++ b/dbms/src/Storages/Transaction/RegionCFDataBase.cpp
@@ -235,7 +235,7 @@ size_t RegionCFDataBase<Trait>::serialize(WriteBuffer & buf) const
 template <typename Trait>
 size_t RegionCFDataBase<Trait>::deserialize(ReadBuffer & buf, RegionCFDataBase & new_region_data)
 {
-    size_t size = readBinary2<size_t>(buf);
+    auto size = readBinary2<size_t>(buf);
     size_t cf_data_size = 0;
     for (size_t i = 0; i < size; ++i)
     {

--- a/dbms/src/Storages/Transaction/TiKVRecordFormat.h
+++ b/dbms/src/Storages/Transaction/TiKVRecordFormat.h
@@ -65,6 +65,7 @@ static const char ASYNC_COMMIT_PREFIX = 'a';
 static const char ROLLBACK_TS_PREFIX = 'r';
 static const char FLAG_OVERLAPPED_ROLLBACK = 'R';
 static const char GC_FENCE_PREFIX = 'F';
+static const char LAST_CHANGE_PREFIX = 'l';
 
 static const size_t SHORT_VALUE_MAX_LEN = 64;
 

--- a/dbms/src/Storages/Transaction/tests/gtest_tikv_keyvalue.cpp
+++ b/dbms/src/Storages/Transaction/tests/gtest_tikv_keyvalue.cpp
@@ -66,6 +66,11 @@ TiKVValue encode_lock_cf_value(
             res.write(s.data(), s.size());
         }
     }
+    {
+        res.write(RecordKVFormat::LAST_CHANGE_PREFIX);
+        RecordKVFormat::encodeUInt64(12345678, res);
+        TiKV::writeVarUInt(87654321, res);
+    }
     return TiKVValue(res.releaseStr());
 }
 

